Sources/NIOConcurrencyHelpers/atomics.swift

```diff
@@ -19,11 +19,17 @@ import Darwin
 fileprivate func sys_sched_yield() {
     pthread_yield_np()
 }
-#else
+#elseif os(Android) || os(Linux)
```

Comment:
Might be better to leave this in an `else` block as a catch-all, and just bring the `elseif os(Windows)` to the top.

Reply:
I think that the catch all is incorrect - why assume that `glibc` is the underlying libc?  Why not `musl`?  or `bionic`?  or `ucrt`?  Explicitly stating the underlying libc is a better approach IMO.

Reply:
Because the decision matrix works out better. If we don't guess, on any unrecognised platform we 100% fail to build. If we do guess, some amount of the time the backing libc actually _will_ be `glibc` and we'll compile. That's a strictly better outcome.

Additionally, I'm deeply unconvinced by the idea that other libcs are going to be exposed as different imports. They could be, but if they are then we're going to have a _really_ hard time adopting new libcs on Linux, which unconditionally assumes `glibc` is the name for libc. I suspect for better or worse we've accidentally frozen the idea that Posix `libc` is imported as `glibc` even when it isn't.

Reply:
I think that we will need to revisit that decision as it doesn't hold.  I would like to be able to build against musl as well.  In the mean time, sure, I can change the condition to prevent a break for the time being.

Reply:
I totally view building against musl as a concrete goal for Swift. I just suspect that we'll end up saying "musl is glibc" rather than forcing the whole Swift on Linux ecosystem to pretend otherwise. That's what C does, after all. :grin:

Reply:
There are header differences, so it will have to be `musl` ... the module layout is just not compatible enough.